src/core/binary_reader.cpp

```diff
@@ -44,15 +44,15 @@ std::string BinaryReader::readString()
 		return std::string();
 	}
 
-	char rawValue[length + 1];
+	std::vector<char> rawValue(length + 1);
```

Comment:
Oh interesting, I missed why you needed to do this. Seems unfortunate as the elements will now get allocated on the heap. Is this due to the windows compiler not supporting a variable length array initializer? 

We could always toggle this via another define, but may be a little overkill to swiss cheese the code for this (unless we were using it in a bunch of other performance critical areas). 

@hermet does this impact your project in any way? Most of the string allocations are fairly small, but there is theoretically no upper limit imposed by the editor so perhaps this heap-based solution is safer anyway...

Reply:
Yeah, it seems that Windows only support fixed-length array initializers... I've not yet deeply tested it to see if it has any impact on performances.

Reply:
> Oh interesting, I missed why you needed to do this. Seems unfortunate as the elements will now get allocated on the heap. Is this due to the windows compiler not supporting a variable length array initializer?
> 
> We could always toggle this via another define, but may be a little overkill to swiss cheese the code for this (unless we were using it in a bunch of other performance critical areas).
> 
> @hermet does this impact your project in any way? Most of the string allocations are fairly small, but there is theoretically no upper limit imposed by the editor so perhaps this heap-based solution is safer anyway...


@luigi-rosso late reply, looks no problem. it works fine.